### PR TITLE
chore(Table) convert demos to ts

### LINKED ISSUE #9447: Convert table react demos to TS

### PR DESCRIPTION
**Pull request by Dominik-Petrik**
**What**: Closes #9447 

~~Includes changes from #9616 and #9618~~

There are a few import errors when yarn start is run I don't know how to solve.

### CODE REVIEW

**Review by jenny-s51 — COMMENTED** [review]
Fixed build, working on updating the remaining demos to TSX

**Review by jenny-s51 — COMMENTED** [review]
Update:

Converted remaining table demos to TSX.

`ColumnManagementWithDraggable` demo requires heavy refactoring and  reimplementation using the new Draggable and Droppable components. Working out the remaining logic so we can also close https://github.com/patternfly/patternfly-react/issues/7471

**Comment by jenny-s51** [comment]
`ColumnManagementWithDraggable` demo has been converted to TS.

As mentioned above, following more investigation, the `ColumnManagementWithDraggable` logic requires a significant overhaul due to the new DragDrop implementation. Work has been started [here](https://github.com/jenny-s51/patternfly-react/tree/colMgmtWithDnDTableDemo).

Following discussion/consensus in standup today with @thatblindgeye and @kmcfaul... the scope of the DragDrop-related refactoring effort is large, and it is out of scope of this original TS conversion issue, therefore it makes more sense to merge in this PR and track the additional work towards #7471 separately. 

Furthermore, additional changes to the DragDrop implementation are currently in progress (see https://github.com/patternfly/patternfly-react/issues/6317).

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
In addition to below, I'm also noticing in the bulk select demo that the styling of the split button toggle is off when checked:

Split button toggle with incorrect styling

It's like this on the staging site as well cc @mcoker looks like it's the `align-self: center` that was added in https://github.com/patternfly/patternfly/pull/5947 ?

File: packages/react-table/src/demos/examples/TableBulkSelect.tsx [hunk 1]
```diff
@@ -6,44 +6,51 @@ import {
...
+import { rows, columns, SampleDataRow } from '@patternfly/react-table/dist/esm/demos/sampleData';
+import { DashboardWrapper } from '@patternfly/react-core/dist/esm/demos/DashboardWrapper';
 
-export const BulkSelectTableDemo = () => {
-  const [isBulkSelectDropdownOpen, setIsBulkSelectDropdownOpen] = React.useState(false);
-  const [bulkSelection, setBulkSelection] = React.useState('');
-  const [page, setPage] = React.useState(1);
-  const [perPage, setPerPage] = React.useState(10);
+export const TableBulkSelect: React.FunctionComponent = () => {
+  const [isBulkSelectDropdownOpen, setIsBulkSelectDropdownOpen] = React.useState<boolean>(false);
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-table/src/demos/examples/TableBulkSelect.tsx]: Good point - updated

**Review by jenny-s51 — COMMENTED** [review]
Thanks @thatblindgeye for your comments! Following our team discussion in standup yesterday about type inference ... since we have `strict` typing on, I added types to areas where ESLint was inferring the `any` type, and made some corresponding updates to logic.

**Comment by wise-king-sullyman** [comment]
@jenny-s51 it looks like there's an a11y error being thrown

File: packages/react-core/src/components/Toolbar/ToolbarToggleGroup.tsx [hunk 2]
```diff
@@ -149,7 +149,7 @@ class ToolbarToggleGroup extends React.Component<ToolbarToggleGroupProps> {
                           aria-label="Show Filters"
                           {...(_isExpanded && { 'aria-expanded': true })}
                           aria-haspopup={_isExpanded && this.isContentPopup()}
-                          aria-controls={expandableContentId}
+                          aria-controls={_isExpanded ? expandableContentId : undefined}
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Toolbar/ToolbarToggleGroup.tsx:152]: As an FYI, Jenny and I discussed this and I opened https://github.com/patternfly/patternfly-react/issues/9827 as an investigation as to whether this should be a temporary fix or not

**Review by tlabaj — APPROVED** [review]

**Pull request merged**